### PR TITLE
Add noindex to validation context

### PR DESCRIPTION
**Pull request by fkpwolf**
[AB#397298](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/397298)

### CODE REVIEW

File: src/docfx/docfx.csproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <ItemGroup>
     <PackageReference Include="Azure.Identity" Version="1.3.0" />
     <PackageReference Include="Azure.Security.KeyVault.Secrets" Version="4.1.0" />
-    <PackageReference Include="docs.validation" Version="1.0.0-beta-00551-f897fd1" />
+    <PackageReference Include="docs.validation" Version="1.0.0-alpha-00561-105fefa" />
```
Comment (fkpwolf) [inline src/docfx/docfx.csproj:17]: I will upgrade version before merge

File: src/docfx/build/metadata/UserMetadata.cs [hunk 2]
```diff
@@ -55,6 +55,7 @@ internal class UserMetadata
         [JsonProperty("_tocRel")]
         public string? TocRel { get; init; }
 
+        [JsonProperty("robots")]
```
Comment (yufeih) [inline src/docfx/build/metadata/UserMetadata.cs:58]: revert? We tend to only mark `JsonProperty` for property names that does not follow the convention.

File: src/docfx/validation/ContentValidator.cs [hunk 3]
```diff
@@ -306,13 +307,16 @@ private bool TryCreateValidationContext(FilePath file, [NotNullWhen(true)] out V
 
         private bool TryCreateValidationContext(FilePath file, bool needMonikers, [NotNullWhen(true)] out ValidationContext? context)
         {
+            var metadata = _metadataProvider.GetMetadata(ErrorBuilder.Null, file);
+            var noindex = metadata.Robots?.Contains("noindex");
```
Comment (yufeih) [inline src/docfx/validation/ContentValidator.cs:311]: Case insensitive compare? both `NOINDEX` and `noindex` are valid.

**Review by yufeih — APPROVED** [review]

**Comment by fkpwolf** [comment]
the diff is excepted since now we ignore noindex markdown so error count decreased. This PR is ready to be merged. @yufeih

**Pull request merged**